directory/directory_test.go

```diff
@@ -0,0 +1,71 @@
+package directory
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCanonicalDockerReference(t *testing.T) {
+	dest := NewDirImageDestination("/path/to/somewhere")
+	_, err := dest.CanonicalDockerReference()
+	assert.NotNil(t, err)
+}
+
+func TestGetPutManifest(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "put-manifest")
```

Comment:
Needs a `defer os.RemoveAll(tmp)` .  And this functions is, I think, long enough for the variable to be called `tmpDir`.